% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{amsmath,amssymb}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math} % this also loads fontspec
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else
  % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\usepackage[margin=1in]{geometry}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\usepackage{booktabs}
\usepackage{caption}
\usepackage{longtable}
\usepackage{colortbl}
\usepackage{array}
\usepackage{anyfontsize}
\usepackage{multirow}
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same}
\hypersetup{
  pdftitle={Descriptive Analysis on Geneomic Alterations in 120 JMP\_Pac Patients using BMI \textgreater= 30 as Obese patient cut-off},
  pdfauthor={Yotsawat Pomyen},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}

\title{Descriptive Analysis on Geneomic Alterations in 120 JMP\_Pac
Patients using BMI \textgreater= 30 as Obese patient cut-off}
\author{Yotsawat Pomyen}
\date{r Sys.Date()}

\begin{document}
\maketitle

This is the descriptive analysis on mutational alterations data from
Caris patient sequencing reports. The data were extracted from XML by
Kasonde using 01.extract\_genomic\_data.Rmd and processed by using
03.complexheatmap.Rmd. The scripts are in Dang-Lab GitHub Repo under
JMP\_PAC folder.

\hypertarget{data-used-in-the-analysis}{%
\subsubsection{Data used in the
analysis:}\label{data-used-in-the-analysis}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  \textbf{Summary Report (sreport\_sheet3)}: This data set provides a
  detailed report of molecular alterations across genes, summarizing the
  types and frequencies of mutations observed in the study cohort.
\item
  \textbf{Clinical Dataset (Masterlist)}: Contains clinical data for all
  patients included in the study. This masterlist links genetic data
  with patient identifiers, clinical outcomes, and other relevant
  clinical parameters.
\end{enumerate}

\hypertarget{masld-diagnostic-criteria}{%
\subsubsection{MASLD diagnostic
criteria}\label{masld-diagnostic-criteria}}

According to \href{https://www.aasld.org/new-masld-nomenclature}{AASLD},
adults with MASLD must have NAFLD (steatosis) and at least one of the
four following conditions:

\begin{itemize}
\tightlist
\item
  Hypertension

  \begin{itemize}
  \tightlist
  \item
    BP \(\ge\) 130/85 mmHg OR
  \item
    specific antihypertensive drug treatment
  \item
    We used \textbf{diagnosis of hypertension} from EMR
  \end{itemize}
\end{itemize}

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{Hypertension} \\ 
\cmidrule(lr){2-3}
NASH/NAFLD Status & N & Y \\ 
\midrule\addlinespace[2.5pt]
N & 21 & 56 \\ 
Y & 7 & 36 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.187\\
\end{minipage}
\endgroup

\begin{itemize}
\tightlist
\item
  Hyperlipidemia

  \begin{itemize}
  \tightlist
  \item
    Plasma Triglyceride \(\ge\) 1.70 mmol {[}150 mg/dL{]} OR
  \item
    Plasma HDL-cholesterol \(\le\) 1.0 mmol {[}40 mg/dL{]} (M)/1.3 mmol
    {[}50 mg/dL{]} (F) OR
  \item
    Lipid lowering treatment
  \item
    We used \textbf{diagnosis of hyperlipidemia} from EMR
  \end{itemize}
\end{itemize}

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{Hyperlipidemia} \\ 
\cmidrule(lr){2-3}
NASH/NAFLD Status & N & Y \\ 
\midrule\addlinespace[2.5pt]
N & 21 & 56 \\ 
Y & 13 & 30 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.674\\
\end{minipage}
\endgroup

\begin{itemize}
\tightlist
\item
  Diabetes

  \begin{itemize}
  \tightlist
  \item
    Fasting serum glucose \(\ge\) 5.6 nmol/L {[}100 mg/dL{]}~OR
  \item
    2-hour-post-load glucose levels \(\ge\) 7.8 nmol/L OR
  \item
    HbA1c \(\ge\) 5.7\% {[}39 nmol/L{]} OR
  \item
    Treatment for DM
  \item
    We used \textbf{diagnosis of Type 1 or Type 2 DM} from EMR,
    excluding pre-DM/NA/unspecified
  \end{itemize}
\end{itemize}

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{Diabetes} \\ 
\cmidrule(lr){2-3}
NASH/NAFLD Status & N & Y \\ 
\midrule\addlinespace[2.5pt]
N & 55 & 22 \\ 
Y & 26 & 17 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.23\\
\end{minipage}
\endgroup

\begin{itemize}
\tightlist
\item
  Obesity

  \begin{itemize}
  \tightlist
  \item
    BMI \(\ge\) 25 OR
  \item
    WC \(\ge\) 94 cm(M)/80 cm(F) OR
  \item
    Ethnicity adjusted
  \item
    In our analysis we used two criteria: \textbf{BMI \(\ge\) 25} and
    \textbf{BMI \(\ge\) 30}
  \end{itemize}
\end{itemize}

Obesity with BMI \(\ge\) 25 \begingroup
\fontsize{12.0pt}{14.4pt}\selectfont \setlength{\LTpost}{0mm}

\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{Obese (BMI >= 25)} \\ 
\cmidrule(lr){2-3}
NASH/NAFLD Status & N & Y \\ 
\midrule\addlinespace[2.5pt]
N & 36 & 41 \\ 
Y & 14 & 29 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.176\\
\end{minipage}
\endgroup
\pagebreak

Obesity with BMI \(\ge\) 30 \begingroup
\fontsize{12.0pt}{14.4pt}\selectfont \setlength{\LTpost}{0mm}

\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{Obese (BMI >= 30)} \\ 
\cmidrule(lr){2-3}
NASH/NAFLD Status & N & Y \\ 
\midrule\addlinespace[2.5pt]
N & 66 & 11 \\ 
Y & 31 & 12 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.0907\\
\end{minipage}
\endgroup

\hypertarget{masld-patient-status}{%
\subsubsection{MASLD Patient Status}\label{masld-patient-status}}

\hypertarget{masld-patient-status-with-bmi-ge-25}{%
\subparagraph{\texorpdfstring{MASLD Patient Status with BMI \(\ge\)
25}{MASLD Patient Status with BMI \textbackslash ge 25}}\label{masld-patient-status-with-bmi-ge-25}}

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{MASLD 25} \\ 
\cmidrule(lr){2-3}
 & n & percent \\ 
\midrule\addlinespace[2.5pt]
N & 77 & 0.6416667 \\ 
Y & 43 & 0.3583333 \\ 
\bottomrule
\end{longtable}
\endgroup

\hypertarget{masld-patient-status-with-bmi-ge-30}{%
\subparagraph{\texorpdfstring{MASLD Patient Status with BMI \(\ge\)
30}{MASLD Patient Status with BMI \textbackslash ge 30}}\label{masld-patient-status-with-bmi-ge-30}}

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{MASLD 30} \\ 
\cmidrule(lr){2-3}
 & n & percent \\ 
\midrule\addlinespace[2.5pt]
N & 78 & 0.65 \\ 
Y & 42 & 0.35 \\ 
\bottomrule
\end{longtable}
\endgroup

MASLD patients between BMI \(\ge\) 25 or 30 is not significantly
different.

\hypertarget{kras-mutation-v-masld}{%
\subsubsection{KRAS Mutation v MASLD}\label{kras-mutation-v-masld}}

\begin{verbatim}
## Warning: The `scale_name` argument of `continuous_scale()` is deprecated as of ggplot2
## 3.5.0.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.
\end{verbatim}

\begin{verbatim}
## Warning: The `trans` argument of `continuous_scale()` is deprecated as of ggplot2 3.5.0.
## i Please use the `transform` argument instead.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.
\end{verbatim}

\begin{verbatim}
## Warning: `unite_()` was deprecated in tidyr 1.2.0.
## i Please use `unite()` instead.
## i The deprecated feature was likely used in the ggmosaic package.
##   Please report the issue at <https://github.com/haleyjeppson/ggmosaic>.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
## generated.
\end{verbatim}

\includegraphics{JMP_Pac_Descriptive_Mutation_Analysis_2024SEP23_files/figure-latex/KRAS nucleotide changes plot-1.pdf}

\includegraphics{JMP_Pac_Descriptive_Mutation_Analysis_2024SEP23_files/figure-latex/KRAS protein changes plot-1.pdf}

\includegraphics{JMP_Pac_Descriptive_Mutation_Analysis_2024SEP23_files/figure-latex/unnamed-chunk-1-1.pdf}
\pagebreak \#\#\# Other Cancer Type Relevant Biomarkers status

\hypertarget{braf}{%
\paragraph{BRAF}\label{braf}}

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{BRAF Aplification} \\ 
\cmidrule(lr){2-3}
MASLD30 Status & Indeterminate & Not Detected \\ 
\midrule\addlinespace[2.5pt]
N & 2 & 75 \\ 
Y & 0 & 42 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.539\\
\end{minipage}
\endgroup
\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rrr}
\toprule
 & \multicolumn{3}{c}{BRAF DNA-seq} \\ 
\cmidrule(lr){2-4}
MASLD30 Status & Likely Pathogenic Variant & Pathogenic Variant & Wild Type \\ 
\midrule\addlinespace[2.5pt]
N & 2 & 1 & 74 \\ 
Y & 0 & 1 & 41 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.788\\
\end{minipage}
\endgroup

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\begin{longtable}{l|r}
\toprule
 & BRAF Fusion \\ 
\cmidrule(lr){2-2}
MASLD30 Status & Not Detected \\ 
\midrule\addlinespace[2.5pt]
N & 78 \\ 
Y & 42 \\ 
\bottomrule
\end{longtable}
\endgroup
\pagebreak

\hypertarget{genomic-signatures}{%
\paragraph{Genomic Signatures}\label{genomic-signatures}}

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rrr}
\toprule
 & \multicolumn{3}{c}{Microsatellite Instability} \\ 
\cmidrule(lr){2-4}
MASLD30 Status & High & No Result & Normal \\ 
\midrule\addlinespace[2.5pt]
N & 1 & 1 & 75 \\ 
Y & 0 & 0 & 42 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  1\\
\end{minipage}
\endgroup

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{Tumor Mutational Burden} \\ 
\cmidrule(lr){2-3}
MASLD30 Status & Low & No Result \\ 
\midrule\addlinespace[2.5pt]
N & 76 & 1 \\ 
Y & 42 & 0 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  1\\
\end{minipage}
\endgroup

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{lrrr}
\toprule
 & \multicolumn{3}{c}{Genomic LoH} \\ 
\cmidrule(lr){2-4}
MASLD30 & High & Low & Normal \\ 
\midrule\addlinespace[2.5pt]
N & 2 & 74 & 1 \\ 
Y & 0 & 42 & 0 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.701\\
\end{minipage}
\endgroup
\pagebreak

\hypertarget{ntrk1}{%
\paragraph{NTRK1}\label{ntrk1}}

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rrr}
\toprule
 & \multicolumn{3}{c}{NTRK1 Amplification} \\ 
\cmidrule(lr){2-4}
MASLD30 Status & Indeterminate & Intermediate & Not Detected \\ 
\midrule\addlinespace[2.5pt]
N & 2 & 6 & 70 \\ 
Y & 0 & 0 & 42 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.0945\\
\end{minipage}
\endgroup

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{lrr}
\toprule
 & \multicolumn{2}{c}{NTRK1 Mutation} \\ 
\cmidrule(lr){2-3}
MASLD25 & VUS & Wild Type \\ 
\midrule\addlinespace[2.5pt]
N & 2 & 73 \\ 
Y & 0 & 43 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.538\\
\end{minipage}
\endgroup

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\begin{longtable}{l|r}
\toprule
 & NTRK1 Fusion \\ 
\cmidrule(lr){2-2}
MASLD30 Status & Not Detected \\ 
\midrule\addlinespace[2.5pt]
N & 78 \\ 
Y & 42 \\ 
\bottomrule
\end{longtable}
\endgroup
\pagebreak

\hypertarget{ret}{%
\paragraph{RET}\label{ret}}

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{RET Amplification} \\ 
\cmidrule(lr){2-3}
MASLD30 Status & Indeterminate & Not Detected \\ 
\midrule\addlinespace[2.5pt]
N & 2 & 76 \\ 
Y & 0 & 42 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.541\\
\end{minipage}
\endgroup

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rrr}
\toprule
 & \multicolumn{2}{c}{RET Mutation} &  \\ 
\cmidrule(lr){2-3}
MASLD30 Status & Likely Benign Variant & VUS & Wild Type \\ 
\midrule\addlinespace[2.5pt]
N & 1 & 2 & 74 \\ 
Y & 0 & 0 & 42 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.701\\
\end{minipage}
\endgroup

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\begin{longtable}{l|r}
\toprule
 & RET Fusion \\ 
\cmidrule(lr){2-2}
MASLD30 Status & Not Detected \\ 
\midrule\addlinespace[2.5pt]
N & 78 \\ 
Y & 42 \\ 
\bottomrule
\end{longtable}
\endgroup
\pagebreak

\hypertarget{atm}{%
\paragraph{ATM}\label{atm}}

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{ATM Deletion} \\ 
\cmidrule(lr){2-3}
MASLD30 Status & Indeterminate & Not Detected \\ 
\midrule\addlinespace[2.5pt]
N & 9 & 68 \\ 
Y & 5 & 37 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  1\\
\end{minipage}
\endgroup

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rrrr}
\toprule
 & \multicolumn{4}{c}{ATM Mutation} \\ 
\cmidrule(lr){2-5}
MASLD30 Status & Likely Pathogenic Variant & Pathogenic Variant & VUS & Wild Type \\ 
\midrule\addlinespace[2.5pt]
N & 1 & 0 & 3 & 72 \\ 
Y & 0 & 2 & 3 & 37 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.163\\
\end{minipage}
\endgroup

\pagebreak

\hypertarget{brac1}{%
\paragraph{BRAC1}\label{brac1}}

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{BRCA1 Deletion} \\ 
\cmidrule(lr){2-3}
MASLD30 Status & Indeterminate & Not Detected \\ 
\midrule\addlinespace[2.5pt]
N & 8 & 69 \\ 
Y & 5 & 37 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.769\\
\end{minipage}
\endgroup

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{BRCA1 Mutation} \\ 
\cmidrule(lr){2-3}
MASLD30 Status & VUS & Wild Type \\ 
\midrule\addlinespace[2.5pt]
N & 2 & 74 \\ 
Y & 1 & 41 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  1\\
\end{minipage}
\endgroup
\pagebreak

\hypertarget{brca2}{%
\paragraph{BRCA2}\label{brca2}}

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{BRCA2 Deletion} \\ 
\cmidrule(lr){2-3}
MASLD30 Status & Indeterminate & Not Detected \\ 
\midrule\addlinespace[2.5pt]
N & 3 & 74 \\ 
Y & 0 & 42 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.551\\
\end{minipage}
\endgroup

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rrr}
\toprule
 & \multicolumn{3}{c}{BRCA2 Mutation} \\ 
\cmidrule(lr){2-4}
MASLD30 Status & Pathogenic Variant & VUS & Wild Type \\ 
\midrule\addlinespace[2.5pt]
N & 0 & 0 & 77 \\ 
Y & 2 & 3 & 38 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.00505\\
\end{minipage}
\endgroup
\pagebreak

\hypertarget{mtap}{%
\paragraph{MTAP}\label{mtap}}

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{MTAP Deletion} \\ 
\cmidrule(lr){2-3}
MASLD30 Status & Indeterminate & Not Detected \\ 
\midrule\addlinespace[2.5pt]
N & 17 & 60 \\ 
Y & 15 & 27 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.132\\
\end{minipage}
\endgroup

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{MTAP Mutation} \\ 
\cmidrule(lr){2-3}
MASLD30 Status & indeterminate & not detected \\ 
\midrule\addlinespace[2.5pt]
N & 17 & 60 \\ 
Y & 4 & 38 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.13\\
\end{minipage}
\endgroup

\hypertarget{nrg1}{%
\paragraph{NRG1}\label{nrg1}}

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{NRG1 Fusion} \\ 
\cmidrule(lr){2-3}
MASLD30 Status & Not Detected & Pathogenic Fusion \\ 
\midrule\addlinespace[2.5pt]
N & 77 & 1 \\ 
Y & 42 & 0 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  1\\
\end{minipage}
\endgroup
\pagebreak

\hypertarget{palb2}{%
\paragraph{PALB2}\label{palb2}}

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{PALB2 Deletion} \\ 
\cmidrule(lr){2-3}
MASLD30 Status & Indeterminate & Not Detected \\ 
\midrule\addlinespace[2.5pt]
N & 7 & 71 \\ 
Y & 7 & 35 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.241\\
\end{minipage}
\endgroup

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{PALB2 Mutation} \\ 
\cmidrule(lr){2-3}
MASLD30 Status & VUS & Wild Type \\ 
\midrule\addlinespace[2.5pt]
N & 1 & 75 \\ 
Y & 1 & 41 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  1\\
\end{minipage}
\endgroup
\pagebreak

\hypertarget{smad4}{%
\paragraph{SMAD4}\label{smad4}}

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rrr}
\toprule
 & \multicolumn{3}{c}{SMAD4 Deletion} \\ 
\cmidrule(lr){2-4}
MASLD30 Status & Deleted & Indeterminate & Not Detected \\ 
\midrule\addlinespace[2.5pt]
N & 1 & 24 & 52 \\ 
Y & 0 & 21 & 21 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.0872\\
\end{minipage}
\endgroup

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rrrr}
\toprule
 & \multicolumn{4}{c}{SMAD4 Mutation} \\ 
\cmidrule(lr){2-5}
MASLD30 Status & Likely Pathogenic Variant & Pathogenic Variant & VUS & Wild Type \\ 
\midrule\addlinespace[2.5pt]
N & 1 & 18 & 2 & 56 \\ 
Y & 1 & 9 & 3 & 29 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  0.615\\
\end{minipage}
\endgroup
\pagebreak

\hypertarget{hla-a}{%
\paragraph{HLA-A}\label{hla-a}}

\begingroup
\fontsize{12.0pt}{14.4pt}\selectfont
\setlength{\LTpost}{0mm}
\begin{longtable}{l|rr}
\toprule
 & \multicolumn{2}{c}{HLA-A} \\ 
\cmidrule(lr){2-3}
MASLD30 Status & Mutated & Normal \\ 
\midrule\addlinespace[2.5pt]
N & 26 & 51 \\ 
Y & 15 & 28 \\ 
\bottomrule
\end{longtable}
\begin{minipage}{\linewidth}
Fisher test p-value =  1\\
\end{minipage}
\endgroup

\hypertarget{there-is-no-hla-b-and-hla-c-mutation.}{%
\paragraph{There is no HLA-B and HLA-C
mutation.}\label{there-is-no-hla-b-and-hla-c-mutation.}}

\end{document}
