---
title: "JMP PAC OncoPrint complex Heatmap"
author: "Kasonde Chewe"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    df_print: paged
---

# Technical
#### last modified: 08/15/2024

Oncoplots are visualizations generated by `ComplexHeatmap` that effectively summarize mutation types across samples. Conceptually, these can be represented as an \( M \times N \) matrix, where \( M \) corresponds to the genes (rows) and \( N \) represents the samples (columns).

In this matrix representation, each element \( a_{ij} \) of the matrix corresponds to a cell in the oncoplot, where \( i \) denotes a specific gene and \( j \) denotes a specific sample. The value at each position \( a_{ij} \), is a string that encodes all possible mutation types observed for gene \( i \) in sample \( j \). These mutation types are typically concatenated into a single string using a delimiter (e.g., ";").

The dataset for these oncoplots is usually prepared by summarizing mutation data, which can be extracted from Mutation Annotation Format (MAF) files or Variant Call Format (VCF) files.

Two primary tools are used to construct these informative figures:

1. **ComplexHeatmap:** For creating heatmaps that can be tailored to display various genetic alterations across a cohort of samples.
2. **Circos:** For generating circular plots that can show relationships and patterns across different chromosomes and samples (not used directly in this script but commonly paired with oncoplots for comprehensive genomic visualizations).

```{r, include=FALSE, echo=FALSE}
# Clear environment variables 
rm(list=ls())
gc()
```


```{r}
# source("02.oncoplot_helpers.R")

# Load libraries
suppressPackageStartupMessages(library(ComplexHeatmap))
suppressWarnings(library(readxl))
suppressWarnings(library(ggplot2))
suppressWarnings(library(forcats))
suppressMessages(library(dplyr))
suppressWarnings(library(tidyverse))
suppressWarnings(library(grid))
```


## Loading Datasets for Mutation Analysis
The datasets loaded for mutation analysis in this project serve various purposes, each contributing uniquely to the comprehensive assessment of genetic alterations and their clinical implications:

1. **Summary Report (sreport_sheet3)**: This dataset provides a detailed report of molecular alterations across genes, summarizing the types and frequencies of mutations observed in the study cohort.

2. **Clinical Dataset (Masterlist)**: Contains clinical data for all patients included in the study. This masterlist links genetic data with patient identifiers, clinical outcomes, and other relevant clinical parameters.

3. **Summary Data for Recurred Patients**:
    - **RecurredPatientMutations**: Details specific mutation events in patients who have experienced recurrence of disease.
    - **RecurredPatientMutations_Summary**: Provides a summarized view of the mutation landscape in recurred patients, focusing on key genetic changes that might be linked to disease progression.

4. **Unfiltered Dataset of Mutations (Mutations_Unfiltered)**: Includes comprehensive mutation data before any filtering for quality or relevance. This dataset allows for initial explorations and assessments of the raw genetic data.

Each dataset is crucial for building a robust analysis framework, aiding in the development of targeted therapies and better understanding of the genetic underpinnings of disease recurrence and progression.

```{r}
# Load Mutation Summaries from CARIS REPORTS
# setwd("/oceanus/collab/InternalJeff/users/hxd052/data/JMP_PAC/datasets")
setwd("C:/Users/yxp177/Documents/Dang-Lab/JMP_PAC")
sreport_sheet3 <- read_xlsx(path = "jmp_pac_mutation_summary_v2.xlsx") # update for final pipeline (mutation summary report)
head(sreport_sheet3)
dim(sreport_sheet3)
colnames(sreport_sheet3)
cat("Mutation Summaries loaded!!!\n")

# Load Main Datasets ----
mst_list <- read_xlsx("JMP_PaC_NAFLD_n502_Extended_Dataset.xlsx")
head(mst_list)
dim(mst_list)
colnames(mst_list)
cat("Main dataset loaded!!!\n")

# Extract sub-dataframe of patients with sequencing
sequenced <- mst_list %>% dplyr::select(`JMP_PAC ID`, `Res_CARIS ID`, 
                                        `Res_SEQ-WTS (Y/N)`, `Res_SEQ-WES (Y/N)`, 
                                        `NASH/NAFLD Status (Y/N)`,
                                        `Hypertension (Y/N)`,
                                        `Hyperlipidemia Status (Y/N)`,
                                        `BMI (kg/m2) before DOS`, 
                                        `DIABETES TYPE (Pre-op)`,
                                        `TNM Stage`,
                                        `TNM-Metastasis`,
                                        `Recurrence (05/2024)? (Y/N)`,
                                        `Recurrence: Distant (Y/N)?`,
                                        `Recurrence: Local (Y/N)?`,
                                        # `Site of Distant Recurrence`,
                                        `Invasion of Lymph Nodes`,
                                        `PANCREATITIS (Y/N)`)
# Filter rows where `Res_CARIS ID` is Y only (patients with WTS sequencing)
wts_sequenced <- sequenced %>% dplyr::filter(`Res_SEQ-WTS (Y/N)` == "Y")
cat("Filtered sequenced patients.\n")

# Load previously saved file paths
# wts_sequenced <- read_xlsx("/excels/summaries/jmp_pac_sequenced.xlsx")
# cat("Loaded XML file paths!!!\n")

## Drop some unnecessary columns 
to.drop <- c('Res_SEQ-WTS (Y/N)','Res_SEQ-WES (Y/N)')
wts_sequenced <- wts_sequenced %>% select(-any_of(to.drop))

## remove NA from NASH/NAFLD status 
wts_sequenced[which(wts_sequenced$`NASH/NAFLD Status (Y/N)` == 'NA'),3] <- 'N'

```

```{r Change DIABETES TYPE to DM Status, include=FALSE}
wts_sequenced <- wts_sequenced %>% mutate(DM = case_when(`DIABETES TYPE (Pre-op)` == "NA" | `DIABETES TYPE (Pre-op)` == "prediabetic" | `DIABETES TYPE (Pre-op)` == "Prediabetic" | `DIABETES TYPE (Pre-op)` == "unspecified" | `DIABETES TYPE (Pre-op)` == "post-pancreatectomy diabetes" ~ "N",
                                                         `DIABETES TYPE (Pre-op)` == "type 1" | `DIABETES TYPE (Pre-op)` == "Type 2" | `DIABETES TYPE (Pre-op)` == 'type 2' ~ "Y"))

```

```{r Obese25, BMI >= 25 cutoff, include=FALSE}
wts_sequenced <- wts_sequenced %>% mutate(`Obese25` = case_when(`BMI (kg/m2) before DOS` < 25 ~ "N", 
                                                                `BMI (kg/m2) before DOS` >= 25 ~ "Y"))

```

```{r MASLD status determination, include=FALSE} 
wts_sequenced <- wts_sequenced %>% mutate(MASLD = case_when(`NASH/NAFLD Status (Y/N)` == "Y" & `Hypertension (Y/N)` == "Y" ~ "Y", 
                                                            `NASH/NAFLD Status (Y/N)` == "Y" & `Hyperlipidemia Status (Y/N)` == "Y" ~ "Y", 
                                                            `NASH/NAFLD Status (Y/N)` == "Y" & `BMI (kg/m2) before DOS` >= 25 ~ "Y",
                                                            `NASH/NAFLD Status (Y/N)` == "Y" & DM == "Y" ~ "Y", 
                                                            `NASH/NAFLD Status (Y/N)` == "N" ~ "N"))

```

```{r TNM-Metastasis remapping, include=FALSE}
wts_sequenced <- wts_sequenced %>% mutate(PreOpMet = case_when(`TNM-Metastasis` == "cM0" ~ "M0",
                                                               `TNM-Metastasis` == "cM1" ~ "M1",
                                                               `TNM-Metastasis` == "cMx" ~ "Mx",
                                                               `TNM-Metastasis` == "Mx" ~ "Mx", 
                                                               `TNM-Metastasis` == "NA" ~ "Mx"))

```

```{r remove denominator in invasion of lymp nodes, include=FALSE} 
wts_sequenced$`Invasion of Lymph Nodes` <- as.numeric(gsub('/.*','',wts_sequenced$`Invasion of Lymph Nodes`))
wts_sequenced <- wts_sequenced %>% mutate(`Lymph node invasion` = case_when(`Invasion of Lymph Nodes` == 0 ~ "N",
                                          `Invasion of Lymph Nodes` == 1 ~ "Y"))
```



## Generating Matrix of Molecular Consequences

### Task Overview
For the final presentation, this section focuses on the creation of a matrix that categorizes molecular consequences of mutations. It is vital for distinguishing between somatic and germline mutation types, which have different implications in the study of genetic diseases and cancer. For our dataset we process only somatic mutations

### Mutation Types Included in Analysis
The data contains a range of mutation types, each classified based on its genetic impact:
- **Frameshift**: Mutations that alter the reading frame of the protein-coding DNA sequence.
- **MISSENSE**: Mutations where the change of a single base pair causes the substitution of a different amino acid in the resulting protein.
- **SPLICING**: Mutations that affect the splicing of mRNA during the processing of the initial transcript.
- **Nonsense**: Mutations that introduce a premature stop codon into the mRNA sequence, leading to truncated, non-functional protein.
- **CODON INSERTION**, **CODON DELETION**: Mutations that either insert or delete nucleotides in the codons of the gene.
- **SYNONYMOUS**: Mutations that change a base pair but do not change the amino acid and hence have no impact on the protein sequence.
- **START LOST**: Mutations that affect the start codon and potentially prevent the initiation of translation.
- **NONCODING**: Mutations that occur outside of the coding sequences of genes.
- **CODON_CHANGE_PLUS_CODON_DELETION**: A complex mutation involving changes and deletions within the codons.

### Filtering Approach
The section also includes a filtering process where mutation types are sorted and analyzed based on specific JMP_PAC IDs. This method allows for targeted analysis of mutations in individual patients or groups, supporting refined interpretations and customized treatments.


```{r}
# Set gene column as row names and remove it from the matrix
jmp_matrix <- sreport_sheet3 %>%
  group_by(gene, jmp_pac_id) %>%
  summarize(mutation_consequence = paste(unique(molecularConsequence), collapse = ";"), .groups = 'drop') %>%
  pivot_wider(names_from = jmp_pac_id, values_from = mutation_consequence, values_fill = list(mutation_consequence = "Normal")) %>%
  filter(!is.na(gene)) %>%  # Remove rows where 'gene' is NA
  column_to_rownames(var = "gene") %>%
  as.matrix()

```




```{r}
# Convert the matrix back to a long dataframe for easier manipulation
jmp_matrix_long <- as.data.frame(as.table(jmp_matrix), stringsAsFactors = FALSE)
colnames(jmp_matrix_long) <- c("gene", "JMP_PAC_ID", "mutation_consequence")

# clean matrix further
jmp_matrix_long_clean <- jmp_matrix_long %>%
  mutate(mutation_consequence = str_remove_all(mutation_consequence, ";NA")) %>%
  mutate(mutation_consequence = str_remove_all(mutation_consequence, "NA;")) %>%
  mutate(mutation_consequence = str_replace(mutation_consequence, "^NA$", "")) %>%
  filter(mutation_consequence != "")

# Remove leading and trailing semicolons in case of any leftovers
jmp_matrix_long_clean <- jmp_matrix_long_clean %>%
  mutate(mutation_consequence = str_trim(mutation_consequence, side = "both")) %>%
  mutate(mutation_consequence = str_remove(mutation_consequence, "^;")) %>%
  mutate(mutation_consequence = str_remove(mutation_consequence, ";$"))

# Clean the mutation_consequence column
jmp_matrix_long_clean <- jmp_matrix_long_clean %>%
  mutate(mutation_consequence = str_replace_all(mutation_consequence, "_", " ")) %>%
  mutate(mutation_consequence = str_to_upper(mutation_consequence))

# Convert the long format back to a wide format matrix embedding combined molecular consequences
final_matrix <- jmp_matrix_long_clean %>%
  pivot_wider(id_cols = gene, names_from = JMP_PAC_ID, values_from = mutation_consequence) %>%
  as.data.frame() %>%
  column_to_rownames(var = "gene") %>%
  as.matrix()

# Ensure that the matrix has no NA values by replacing NA with "Normal" or any other placeholder
final_matrix[is.na(final_matrix)] <- "NORMAL"

```



```{r EXTRACT TOP 30 MUTATED GENES}
# Get top 30 mutated genes
# Count the number of mutations (non-"Normal" and non-empty) for each gene
mutation_counts <- apply(final_matrix, 1, function(x) sum(x != "NORMAL" & x != ""))
top_genes <- sort(mutation_counts, decreasing = TRUE)[1:30]
names_top_genes <- names(top_genes)
matrix_top30 <- final_matrix[names_top_genes, ]

# Calculate mutation frequencies (non-"Normal" and non-NA entries)
mutation_frequencies <- colSums(matrix_top30 != "NORMAL" & !is.na(matrix_top30))

# Sort columns by these frequencies in descending order
sorted_column_indices <- order(mutation_frequencies, decreasing = TRUE)
sorted_final_matrix <- matrix_top30[, sorted_column_indices]

# Clean up naming
sorted_final_matrix[sorted_final_matrix == "NORMAL"] <- ""
sorted_final_matrix[is.na(sorted_final_matrix)] <- ""
```


```{r extract matrix with MASLD samples only and non-MASLD samples only, include=FALSE}
masld.samples <- as.data.frame(wts_sequenced[which(wts_sequenced$MASLD == "Y"),1])[,1]
masld_matrix <- sorted_final_matrix[,masld.samples]
nonmasld.samples <- as.data.frame(wts_sequenced[which(wts_sequenced$MASLD == "N"),1])[,1]
nonmasld_matrix <- sorted_final_matrix[,nonmasld.samples]
masld.data <- wts_sequenced %>% select(`JMP_PAC ID`,
                                       ## metabolic features
                                       MASLD,
                                       DM,
                                       `Hyperlipidemia Status (Y/N)`,
                                       `Hypertension (Y/N)`,
                                       Obese25,
                                       `PANCREATITIS (Y/N)`,
                                       ## tumor features
                                       `TNM Stage`,
                                       `PreOpMet`,
                                       `Lymph node invasion`,
                                       `Recurrence (05/2024)? (Y/N)`,
                                       `Recurrence: Distant (Y/N)?`,
                                       `Recurrence: Local (Y/N)?`
                                       )
masld.data$MASLD <- as.factor(masld.data$MASLD)
colnames(masld.data) <- c('JMP_PAC ID','MASLD Status','Diabetes','Hyperlipidemia Status','Hypertension Status','Obese','Pancreatitis Status','TNM Stage', "TNM-Metastasis",'Lymph Node Invasion', 'Recurrence Status', 'Distant','Local')
masld.data <- as.data.frame(masld.data)
rownames(masld.data) <- masld.data[,1]
masld.data <- masld.data[,-1]
```

```{r move MISSENSE to behind other alterations, include=FALSE}
sorted_matrix2 <- sorted_final_matrix
for (i in 1:length(sorted_final_matrix[1,])){
  sorted_matrix2[,i] <- gsub("(.*);MISSENSE","MISSENSE;\\1", sorted_matrix2[,i])
}

```


```{r generate oncoplot, fig.width=15, fig.height=11 include=FALSE}
source('02.oncoplot_helpers.R')
pdf('JMP_PAC_oncoplot_120patients_top30genes_3.pdf', width = 24, height = 11)
generate_oncoplot(sorted_matrix2, c_split = masld.data$`MASLD Status`)
dev.off()
```



The plot below is an oncplot that summarizes the mutation alterations, tumor mutation burden and frequency of mutations per gene. 
This report includes a summary report for the current 120 patients who have recieved whole exome sequencing. 
Though unclear, variant calling is assumed to be conducted against hg38 as the "normal" reference. A custom variant calling pipeline is under works

```{r SAVE ONCOPLOT, fig.width=24, fig.height=11}
pdf("../figures/oncoprint/oncoplot_n120_jmppac_all_patients_top30_mutated_genes.pdf", width = 24, height = 11)
generate_oncoplot(sorted_final_matrix, "Fig-1: JMP PAC Molecular Profile PDAC Oncoplot n=120 ")
dev.off()
```

```{r oncoplot with annotation, fig.width=24, fig.height=11}
pdf("Top_30-gene_clinical_anno_metabolic_tumor_features.pdf", width = 24, height = 11)
source('02.oncoplot_helpers.R')
generate_oncoplot_with_anno(sorted_matrix2, c_split = masld.data$`MASLD Status`, title = "JMP PAC Top 30-Gene Alterations with metabolic and tumor features, n=120")
dev.off()

```


```{r core vs non-core DNA repair genes oncoplot, include=FALSE}
repair_genes <- read_xlsx(path = "DNA Repair Genes.xlsx")
to.drop <- c('Description', 'Chromosome Location (BED format)')
repair_genes <- repair_genes %>% select(-any_of(to.drop))

## find the overlapped genes between final_matrix (Caris DNA-seq data) and list of DNA repiar genes
nonoverlap_genes <- setdiff(repair_genes$Gene, rownames(final_matrix))
repair_genes <- repair_genes %>% filter(Gene != nonoverlap_genes)
repair_genes <- as.data.frame(repair_genes)
rownames(repair_genes) <- repair_genes$Gene
repair_final_matrix <- final_matrix[repair_genes$Gene,]
repair_final_matrix[repair_final_matrix == "NORMAL"] <- ""
```

```{r}

pdf("DNA_repair_genes_metabolic_turmor_features.pdf", width = 24, height = 11)
source('02.oncoplot_helpers.R')
generate_oncoplot_with_anno(repair_final_matrix, c_split = masld.data$`MASLD Status`, r_split = repair_genes$Type, title = "JMP PAC Top Core and non-Core DNA Repair Genes Alterations with metabolic and tumor features, n=120")
dev.off()


```

```{r various DNA repair genes oncoplot, include=FALSE}
dna <- "repair_genes.xlsx"

dna_repair <- dna %>% 
  excel_sheets() %>% 
  set_names() %>% 
  map(read_xlsx, path = dna, col_names = F)

sheet_name <- names(dna_repair)
sheet_name <- sheet_name[-length(sheet_name)]

## function for mapping repair genes 
orepair <- function(x){
  temp <- x$`...1`
  overlap_genes <- intersect(temp,rownames(final_matrix))
  temp <- final_matrix[overlap_genes,]
  ifelse (dim(temp)[1] > 0, temp <- temp[temp == "NORMAL"] <-"",print('0'))
  # ifelse (dim(temp)[1] > 0, {temp <- temp[temp == "NORMAL"] <-"";return(temp)},next)
}
nonoverlap_genes <- setdiff(repair_genes$Gene, rownames(final_matrix))
repair_genes <- repair_genes %>% filter(Gene != nonoverlap_genes)
repair_genes <- as.data.frame(repair_genes)
rownames(repair_genes) <- repair_genes$Gene
repair_final_matrix <- final_matrix[repair_genes$Gene,]
repair_final_matrix[repair_final_matrix == "NORMAL"] <- ""

```


# The codes after this are not used in the analysis

```{r GENERATE ONCOPLOT, fig.width=24, fig.height=11}
generate_oncoplot(sorted_final_matrix, "Fig-1: JMP PAC Molecular Profile PDAC Oncoplot n=120 ")
```

## Extract Risk Factors 
PDAC is associated 
- Diabetes "DIABETIC STATUS (Y/N)"
- Pancreatitis "PANCREATITIS (Y/N)" 
- Pancreatitis "PANCREATITIS (chronic / acute/ NA)"
- Hyperlipidemia ""Hyperlipidemia Status (Y/N)"
- Fatty Liver Disease "NASH/NAFLD Status (Y/N)"
- "Living? (05/2024 Review) (Y/N)"
- Hypertension "Hypertension (Y/N)"
- Obese 


```{r}
library(dplyr)
# Set options to display 4 digits
options(digits = 4)

# Set BMI to numeric data type 
# `BMI (kg/m2) at DOS` is not in the data
# `BMI (kg/m2) before DOS`
mst_list$`BMI (kg/m2) before DOS` <- as.numeric(mst_list$`BMI (kg/m2) before DOS`)
mst_list$`Hyperlipedimia Status (Values)` <- as.numeric(mst_list$`Hyperlipedimia Status (Values)` )
# Define a function to categorize obesity based on BMI
categorize_obesity <- function(bmi) {
  ifelse(bmi >= 30, "Y", "N")
}

# Apply the function to create a new column in the mst_list dataframe
mst_list <- mst_list %>%
  mutate(Obesity = categorize_obesity(`BMI (kg/m2) before DOS`))

# Create dataframe of JMP PAC patients by risk factors
risk_factors <- mst_list %>%
  dplyr::select(`JMP_PAC ID`, 
         Diabetes = `DIABETIC STATUS (Y/N)`,
         Pancreatitis1 = `PANCREATITIS (Y/N)`,
         #Pancreatitis2 = `PANCREATITIS (chronic / acute/ NA)`,
         Hyperlipidemia = `Hyperlipidemia Status (Y/N)`,
         HLD = `Hyperlipedimia Status (Values)`,
         Fatty_Liver_Disease = `NASH/NAFLD Status (Y/N)`,
         #Living_Status = `Living? (05/2024 Review) (Y/N)`,
         Hypertension = `Hypertension (Y/N)`) %>%
         #Obesity,
         #BMI = `BMI (kg/m2) before DOS`) %>%
  filter(`JMP_PAC ID` %in% colnames(jmp_matrix)) %>%
  arrange(factor(`JMP_PAC ID`, levels = colnames(jmp_matrix)))

risk_factors$Fatty_Liver_Disease[is.na(risk_factors$Fatty_Liver_Disease)] <- "N"

# Display the dimensions and summary of the new dataframe
dim(risk_factors)
head(risk_factors)
summary(risk_factors)

```

```{r, fig.width=24, fig.height=11}
pdf("../figures/oncoprint/figa1_oncoplot_n120_jmppac_all_patients_top30_mutated_genes.pdf", width = 22, height = 11)
source('02.oncoplot_helpers.R')
generate_oncoplot_with_anno(sorted_final_matrix, "FigA1: JMP PAC Molecular Profile PDAC Oncoplot n=120 ")
dev.off()
```

```{r}
generate_oncoplot_with_anno(sorted_final_matrix, "FigA1: JMP PAC Molecular Profile PDAC Oncoplot n=120 ")
```



- Stage "TNM-Stage"


```{r}
# TO-DO LIST
# extract TNM stage 
# extract KRAS mutation status
```





# DIVIDE PATIENTS BY FATTY LIVER DISEASE STATUS
```{r}
FL <- wts_sequenced[wts_sequenced$`NASH/NAFLD Status (Y/N)` == "Y", ]
tally_fl <- dim(FL)[1]
cat("\n Number of Patients with Fatty Liver Disease:", tally_fl)


nFL <- wts_sequenced[wts_sequenced$`NASH/NAFLD Status (Y/N)` != "Y" , ]
tally_nfl <- dim(nFL)[1]
cat("\n Number of Patients without Fatty Liver Disease:", tally_nfl)


# PIE CHART FOR PATIENTS WITH AND WITHOUT FLD
data <- c(tally_fl, tally_nfl)
labels <- c("NAFLD", "non-NAFLD")

piepercent<- round(100 * data / sum(data), 1)
# Plot the chart.
pie(data, labels = piepercent,
    main = "Proportion of Patients with NALFD vs non-NAFLD", col = rainbow(length(labels)))
legend("topright", labels ,
                    cex = 0.5, fill = rainbow(length(labels)))

```
## Dividing Patients into two groups
1. NAFLD patients (43)
2. non-NAFLD patients (77), note that this includes 1 unknown who required review

```{r}
sorted_final_df <- as.data.frame(sorted_final_matrix)

# NAFLD 
nafld_matrix <- sorted_final_df %>%
  dplyr::select(one_of(FL$`JMP_PAC ID`)) %>%
  as.matrix()
# View the selected matrix
dim(nafld_matrix)


# non-NAFLD 
nfld_matrix <- sorted_final_df %>%
  dplyr::select(one_of(nFL$`JMP_PAC ID`)) %>%
  as.matrix()
# View the selected matrix
dim(nfld_matrix)

```



```{r GENERATE NAFLD ONCOPLOT, fig.width=24, fig.height=11}
generate_oncoplot(nafld_matrix, "Fig-2: NAFLD Profile PDAC Oncoplot n=43")
```

```{r SAVE NAFLD ONCOPLOT, fig.width=24, fig.height=11}
pdf("../figures/oncoprint/oncoplot_n43_jmppac_NAFLD_patients_top30_mutated_genes.pdf", width = 24, height = 11)
generate_oncoplot(nafld_matrix, "Fig-2: NAFLD Profile PDAC Oncoplot n=43")
dev.off()
```




```{r GENERATE non-NAFLD ONCOPLOT, fig.width=24, fig.height=11}
generate_oncoplot(nfld_matrix, "Fig-3: non-NAFLD Profile PDAC Oncoplot n=77")
```

```{r SAVE non-NAFLD ONCOPLOT, fig.width=24, fig.height=11}
pdf("../figures/oncoprint/oncoplot_n77_jmppac_NAFLD_patients_top30_mutated_genes.pdf", width = 24, height = 11)
generate_oncoplot(nfld_matrix, "Fig-3: Non-NAFLD Profile PDAC Oncoplot n=77")
dev.off()
```




# KRAS Mutations Overview

The Kirsten rat sarcoma viral oncogene homolog (KRAS) is one of the most commonly mutated genes in cancer, particularly in pancreatic ductal adenocarcinoma (PDAC), non-small-cell lung cancer (NSCLC), and colorectal cancer. These mutations lead to abnormal cell growth and are key drivers in the progression of these cancers.

The table below summarizes the most frequent KRAS mutations, including their codon changes, base changes, and the resulting amino acid substitutions. Understanding these mutations is crucial for the development of targeted therapies, which are increasingly becoming a focus in the treatment of cancers associated with KRAS mutations.

#### Table 1: KRAS Mutations Summary

| Mutation Type | Codon Change | Base Change | Amino Acid Change | Cancer Types Associated      |
| ------------- | ------------ | ----------- | ----------------- | ---------------------------- |
| **G12D**      | GGT > GAT    | G > A       | Gly > Asp          | Pancreatic, Colorectal, Lung |
| **G12V**      | GGT > GTT    | G > T       | Gly > Val          | Pancreatic, Colorectal, Lung |
| **G12C**      | GGT > TGT    | G > T       | Gly > Cys          | Lung, Colorectal             |
| **G12R**      | GGT > CGT    | G > C       | Gly > Arg          | Pancreatic, Colorectal       |
| **G12A**      | GGT > GCT    | G > C       | Gly > Ala          | Colorectal                   |
| **G12S**      | GGT > AGT    | G > A       | Gly > Ser          | Colorectal                   |
| **G13D**      | GGC > GAC    | G > A       | Gly > Asp          | Colorectal                   |
| **Q61H**      | CAA > CAC    | C > A       | Gln > His          | Lung, Melanoma               |
| **Q61L**      | CAA > CTA    | C > T       | Gln > Leu          | Various Cancers              |
| **Q61R**      | CAA > CGA    | C > G       | Gln > Arg          | Various Cancers              |

**Notes:**

- **Cancer Association:** The most frequent KRAS mutations, such as G12D, G12V, and G12C, are particularly common in pancreatic, lung, and colorectal cancers.
  
- **Codon Changes:** These refer to single nucleotide substitutions that alter the codon, leading to different amino acids being incorporated into the KRAS protein.

- **Amino Acid Changes:** These mutations often cause the KRAS protein to remain in an active "on" state, promoting uncontrolled cell growth and contributing to cancer progression.

The following section provides a descriptive summary of the types of KRAS mutations and their respective base substitutions, highlighting their significance in our patient cohort.

```{r}
library(dplyr)
library(ggprism)
library(ggrepel)
library(ggpubr)
library(moonBook)
library(webr)

# Select only KRAS genes
KRAS_mutation <- sreport_sheet3[sreport_sheet3$gene == "KRAS" & sreport_sheet3$result != "Amplification Not Detected", ]
KRAS_df <- KRAS_mutation %>% 
  drop_na(jmp_pac_id) 


# Clean data
KRAS_df$hgvsProteinChange <- as.character(KRAS_df$hgvsProteinChange)
KRAS_df$hgvsProteinChange[is.na(KRAS_df$hgvsProteinChange)] <- "wildtype"
KRAS_df$hgvsProteinChange <- gsub("^p\\.", "", KRAS_df$hgvsProteinChange)

# Clean Base Change Representation
# Note original representation e.g. c.35G>T is Human Genome Variation Society (HGVS) presentation c. = coding DNA level, 35 is position
# c.35G>T becomes G>T
KRAS_df$hgvsCodingChange[is.na(KRAS_df$hgvsCodingChange)] <- ""
KRAS_df$hgvsCodingChange <- gsub("^c\\.[0-9]+", "", KRAS_df$hgvsCodingChange) # std base change

# Count the occurrences of each KRAS mutation type including hgvsCodingChange
# That is protein change and base change
mutation_counts <- KRAS_df %>%
   dplyr::count(hgvsProteinChange, hgvsCodingChange) %>%
   mutate(percentage = n / sum(n) * 100)

```



```{r VIEW PIE CHART KRAS MUTATION FREQUENCIES n120, fig.width=8, fig.height=8}
PieDonut(mutation_counts, aes(pies = hgvsCodingChange, donuts=hgvsProteinChange, count = n), 
         ratioByGroup = FALSE,
         showRatioThreshold = 0,
         pieLabelSize = 3,
         donutLabelSize = 5,
        showPieName = FALSE,
        r0=2,
        r2=2)   
```

```{r SAVE PIE CHART KRAS MUTATION FREQUENCIES n120, fig.width=8, fig.height=8}
options(digits=2)
# Create a Donut Chart
pdf("../figures/oncoprint/pie_chart_kras_mutations_freq_n120_wes.pdf", width = 8, height = 8)
PieDonut(mutation_counts, aes(pies = hgvsCodingChange, donuts=hgvsProteinChange, count = n), 
         ratioByGroup = FALSE,
         showRatioThreshold = 0,
         pieLabelSize = 3,
         donutLabelSize = 5,
        showPieName = FALSE,
        r0=2,
        r2=2)                   
dev.off()

```




```{r V2 SAVE PIE CHART KRAS MUTATION FREQUENCIES & BASE CHANGE n120, fig.width=8, fig.height=8}
pdf("../figures/oncoprint/pie_chart_kras_mutations_and_base_change_freq_n120_wes.pdf", width = 8, height = 8)
PieDonut(mutation_counts, aes(pies = hgvsCodingChange, donuts=hgvsProteinChange, count = n), 
         ratioByGroup = FALSE,
         showRatioThreshold = 0,
         pieLabelSize = 4,
         donutLabelSize = 5,
         showPieName = FALSE,
         showRatioPie=F,
         r0=0.3,r1=0.9,r2=1.3)   
dev.off()
```




```{r V2 VIEW PIE CHART KRAS MUTATION FREQUENCIES & BASE CHANGE n120, fig.width=8, fig.height=8}
PieDonut(mutation_counts, aes(pies = hgvsCodingChange, donuts=hgvsProteinChange, count = n), 
         ratioByGroup = FALSE,
         showRatioThreshold = 0,
         pieLabelSize = 4,
         donutLabelSize = 5,
         showPieName = FALSE,
         showRatioPie=F,
         r0=0.3,r1=0.9,r2=1.3)   
```

## KRAS MUTATIONS FATTY LIVER vs NON-FATTY LIVER DISEASE PATIENTS 
```{r}
# Select Patients with NAFLD
KRAS_filtered_NAFLD <- KRAS_df %>%
  filter(jmp_pac_id %in% FL$`JMP_PAC ID`)
dim(KRAS_filtered_NAFLD)
head(KRAS_filtered_NAFLD)

# Select Patients with no NAFLD
KRAS_filtered_nFLD <- KRAS_df %>%
  filter(jmp_pac_id %in% nFL$`JMP_PAC ID`)
dim(KRAS_filtered_nFLD)
head(KRAS_filtered_nFLD)

# ------------------------- fatty liver n=43 ------------------------- 
# Clean data
KRAS_filtered_NAFLD$hgvsProteinChange <- as.character(KRAS_filtered_NAFLD$hgvsProteinChange)
KRAS_filtered_NAFLD$hgvsProteinChange[is.na(KRAS_filtered_NAFLD$hgvsProteinChange)] <- "wildtype"
KRAS_filtered_NAFLD$hgvsProteinChange <- gsub("^p\\.", "", KRAS_filtered_NAFLD$hgvsProteinChange)

# Clean Base Change Representation
KRAS_filtered_NAFLD$hgvsCodingChange[is.na(KRAS_filtered_NAFLD$hgvsCodingChange)] <- ""
KRAS_filtered_NAFLD$hgvsCodingChange <- gsub("^c\\.[0-9]+", "", KRAS_filtered_NAFLD$hgvsCodingChange) # std base change

# Count the occurrences of each KRAS mutation type including hgvsCodingChange
mutation_counts_fl <- KRAS_filtered_NAFLD %>%
   dplyr::count(hgvsProteinChange, hgvsCodingChange) %>%
   mutate(percentage = n / sum(n) * 100)

# --------------------------------------------------------------------


# ------------------------- non-fatty liver n=77 unique ------------------------- 
# Clean data
KRAS_filtered_nFLD$hgvsProteinChange <- as.character(KRAS_filtered_nFLD$hgvsProteinChange)
KRAS_filtered_nFLD$hgvsProteinChange[is.na(KRAS_filtered_nFLD$hgvsProteinChange)] <- "wildtype"
KRAS_filtered_nFLD$hgvsProteinChange <- gsub("^p\\.", "", KRAS_filtered_nFLD$hgvsProteinChange)

# Clean Base Change Representation
KRAS_filtered_nFLD$hgvsCodingChange[is.na(KRAS_filtered_nFLD$hgvsCodingChange)] <- ""
KRAS_filtered_nFLD$hgvsCodingChange <- gsub("^c\\.[0-9]+", "", KRAS_filtered_nFLD$hgvsCodingChange) # std base change

# Count the occurrences of each KRAS mutation type including hgvsCodingChange
mutation_counts_nfl <- KRAS_filtered_nFLD %>%
   dplyr::count(hgvsProteinChange, hgvsCodingChange) %>%
   mutate(percentage = n / sum(n) * 100)

# --------------------------------------------------------------------

```

```{r V3 GENERATE PIE CHARTS NAFLD n43 and non-NAFLD n77, fig.width=13, fig.height=10}
p1 <- PieDonut(mutation_counts_fl, aes(pies = hgvsCodingChange, donuts=hgvsProteinChange, count = n), 
         ratioByGroup = FALSE,
         showRatioThreshold = 0,
         pieLabelSize = 3,
         title = "NAFLD",
         donutLabelSize = 5,
        showPieName = FALSE,
        r0=2,
        r2=2)


p2 <- PieDonut(mutation_counts_nfl, aes(pies = hgvsCodingChange, donuts=hgvsProteinChange, count = n), 
         ratioByGroup = FALSE,
         showRatioThreshold = 0,
         pieLabelSize = 3,
         title = "non-NAFLD",
         donutLabelSize = 5,
        showPieName = FALSE,
        r0=2,
        r2=2) 
```

```{r VIEW PIE CHARTS NAFLD n43 and non-NAFLD n77, fig.width=13, fig.height=10}
ggarrange(p1,p2, ncol = 2)

```


```{r}
pdf("../figures/oncoprint/pie_chart_FLvsNFLD_kras_mutations_and_base_change_freq_n43vsn77_wes.pdf", width = 13, height = 10)
ggarrange(p1,p2, ncol = 2)
dev.off()
```



```{r}
sessioninfo()
```


